strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/InProjectTestReporterLoader.java

```diff
@@ -38,11 +47,12 @@
                         }
                     })
                     .map(is -> {
-                        try {
-                            return SurefireReader.loadTestResults(is);
-                        } catch (XMLStreamException e) {
-                            return null;
+                        final Optional<TestResultParser> testResultParser = javaSPILoader.onlyOne(TestResultParser.class, trp -> "junit".equals(trp.type()));
```

Comment:
Well it is an easy and clean way to not populate service logic across the code. I mean if not means to get an iterable iterate over it find if it is the type we want, throw exception if required and so on so far. One way should be just create a method that receives the type, but a predicate is not so much complex nor add complexity and allows in the future to do other kind of filters.